src/Creator/InvoiceCreator.php

```diff
@@ -35,16 +35,16 @@ public function __construct(
     ) {
     }
 
-    public function __invoke(string $orderNumber, \DateTimeInterface $dateTime): void
+    public function __invoke(int $orderId, \DateTimeInterface $dateTime): void
     {
         /** @var OrderInterface $order */
-        $order = $this->orderRepository->findOneByNumber($orderNumber);
+        $order = $this->orderRepository->find($orderId);
```

Comment:
```suggestion
        /** @var OrderInterface|null $order */
        $order = $this->orderRepository->findOneBy(['number' => $orderNumber]);
        Assert::notNull($order, sprintf('Order with number "%s" does not exist.', $orderNumber));
```

Reply:
The Behat test is maybe missing something, in my current projet I have to prepare my order doing this :

```php
$stateMachine = $this->stateMachineFactory->get($order, OrderTransitions::GRAPH);

$stateMachine->apply(OrderTransitions::TRANSITION_CREATE);
```

save it to the database and complete the order (`OrderCheckoutTransitions::TRANSITION_COMPLETE`).